ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/tasks/CompleteBlocksTaskTest.java

```diff
@@ -79,6 +95,115 @@ public void shouldCompleteWithoutPeersWhenAllBlocksAreEmpty() {
     assertThat(task.run()).isCompletedWithValue(blocks);
   }
 
+  @Test
+  public void shouldCreateCorrectEmptyBlockWhenWithdrawalsAreEnabled() {
+    final ProtocolSchedule mockProtocolSchedule = Mockito.mock(ProtocolSchedule.class);
+    final ProtocolSpec mockParisSpec = Mockito.mock(ProtocolSpec.class);
+    final ProtocolSpec mockShanghaiSpec = Mockito.mock(ProtocolSpec.class);
+    final WithdrawalsProcessor mockWithdrawalsProcessor = Mockito.mock(WithdrawalsProcessor.class);
+
+    final BlockHeader header1 =
+        new BlockHeaderTestFixture().number(1).withdrawalsRoot(null).buildHeader();
+    final BlockHeader header2 =
+        new BlockHeaderTestFixture().number(2).withdrawalsRoot(Hash.EMPTY_TRIE_HASH).buildHeader();
+
+    when(mockProtocolSchedule.getByBlockHeader((eq(header1)))).thenReturn(mockParisSpec);
+    when(mockParisSpec.getWithdrawalsProcessor()).thenReturn(Optional.empty());
+    when(mockProtocolSchedule.getByBlockHeader((eq(header2)))).thenReturn(mockShanghaiSpec);
+    when(mockShanghaiSpec.getWithdrawalsProcessor())
+        .thenReturn(Optional.of(mockWithdrawalsProcessor));
+
+    final Block block1 =
+        new Block(header1, new BlockBody(Collections.emptyList(), Collections.emptyList()));
+    final Block block2 =
+        new Block(
+            header2,
+            new BlockBody(
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Optional.of(Collections.emptyList())));
+
+    final List<Block> expectedBlocks = asList(block1, block2);
+    final EthTask<List<Block>> task =
+        CompleteBlocksTask.forHeaders(
+            mockProtocolSchedule,
+            ethContext,
+            List.of(header1, header2),
+            maxRetries,
+            new NoOpMetricsSystem());
+    assertThat(task.run()).isCompletedWithValue(expectedBlocks);
+  }
+
+  @Test
+  public void shouldCompleteWhenBlocksAreEmptyExceptForWithdrawals() {
```

Comment:
Is this testing that when blocks are empty but have withdrawals they are still downloaded? i.e. that they are considered empty blocks. Was a bit confused from the test name

Reply:
> Is this testing that when blocks are empty but have withdrawals they are still downloaded?
yes, i.e. that they are _not_ considered empty blocks like they were being before.

Any name suggestion? This was an modification of the existing test's name: `shouldCompleteWithoutPeersWhenAllBlocksAreEmpty`. Agree the new one could be better